core/src/main/java/com/sequenceiq/cloudbreak/service/decorator/ClusterDecorator.java

```diff
@@ -179,18 +179,19 @@ private void prepareRds(Cluster subject, IdentityUser user, ClusterRequest reque
         if (request.getRdsConfigJsons() != null) {
             for (RDSConfigRequest requestRdsConfig : request.getRdsConfigJsons()) {
                 RDSConfig rdsConfig = conversionService.convert(requestRdsConfig, RDSConfig.class);
+                rdsConfig.setOrganization(stack.getOrganization());
                 rdsConfig.setPublicInAccount(stack.isPublicInAccount());
-                rdsConfig = rdsConfigService.createIfNotExists(user, rdsConfig);
+                rdsConfig = rdsConfigService.createIfNotExists(rdsConfig);
                 subject.getRdsConfigs().add(rdsConfig);
             }
         }
         Optional.of(request.getRdsConfigNames())
-                .ifPresent(confs -> confs.forEach(confName -> subject.getRdsConfigs().add(rdsConfigService.getPublicRdsConfig(confName, user))));
+                .ifPresent(confs -> confs.forEach(confName -> subject.getRdsConfigs().add(rdsConfigService.getByName(confName))));
```

Comment:
it should be ```rdsConfigService.getByNameForOrganizationId(confName, stack.getOrganization().getId())```

Reply:
rdsConfigService is misused in other places too. always use getByNameForOrganizationId or getByNameForOrganization if there is an organizationId available in the scope.